packages/lib/src/interactions/SelectToZoom.tsx

```diff
@@ -17,30 +16,33 @@ interface Props extends CommonInteractionProps {
 function SelectToZoom(props: Props) {
   const { keepRatio, ...interactionProps } = props;
 
-  const { dataToWorld } = useAxisSystemContext();
+  const context = useAxisSystemContext();
   const moveCameraTo = useMoveCameraTo();
 
   const { width, height } = useThree((state) => state.size);
   const camera = useThree((state) => state.camera);
 
-  const { xVisibleDomain, yVisibleDomain } = useCameraState(
-    getVisibleDomains,
-    []
-  );
-  const dataRatio = Math.abs(
-    (xVisibleDomain[1] - xVisibleDomain[0]) /
-      (yVisibleDomain[1] - yVisibleDomain[0])
-  );
+  function getDataRatio() {
+    const { xVisibleDomain, yVisibleDomain } = getVisibleDomains(
+      camera,
+      context
+    );
+    return Math.abs(
+      (xVisibleDomain[1] - xVisibleDomain[0]) /
+        (yVisibleDomain[1] - yVisibleDomain[0])
+    );
+  }
```

Comment:
Similarly, we don't need to re-render and recompute the data ratio every time the visible domains change. Instead, we can compute the visible domains and data ratio only when needed - i.e. when drawing a selection rectangle (in the render prop of `SelectionTool`) and when completing a selection (in `onSelectionEnd`).